src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -229,6 +229,22 @@ static term create_pair(Context *ctx, term term1, term term2)
     return ret;
 }
 
+// term `error` can be an atom or a `const char *const ATOM_STR`
+static term error_tuple_maybe_gc(Context *ctx, term error)
+{
+    term error_atom;
+    if (!term_is_atom(error)) {
```

Comment:
Here you are making some really strong assumptions about term internal format. Also we don't have any control about atom strings alignment, so it might be an atom as a consequence of some small changes.
Let's remember the following test is performed: `((t & 0x3F) == 0xB)`,

Reply:
I really don’t understand this comment.  Can you say more about what you are getting after here?  Are you saying it is invalid to assume that atoms are not GC’d?  We do this all over the place in the codebase.

Reply:
I’m glad I wasn’t the only one a bit confused by this comment, based on the follow up comment below, I do understand that this was not a safe implementation, which I fixed using the suggestion below.

Reply:
What I actually meant is that atom strings are like any other string, just pointer to some memory area that might be at any address. Those pointers are not valid tagged values so the previous implementation was invalid.